### ISSUE TITLE
Where to "inject pose data"

### ISSUE THREAD

**Issue by jbkarle** [marian-margeta/gait-recognition#3]
Hi Marían and thanks for sharing your code!.
I'm a newbie to deep learning, so also thanks for the advise!. I look forward to use an external pose estimator with some different capacities, but in the end I'll finish having the same joints & joints coords for each frame. Where should be feed that data, in order to reuse all your code (feature extraction.. gait identification vector), except for the native pose estimator you used!

Thank you!.

**Comment by marian-margeta** [comment]
Hello @jbkarle,
well, if you already have coordinates of the joints for each frame, you should pre-process the data in the same way as the first subnetwork for pose estimation. 

So you can reuse the function [generate_output](https://github.com/marian-margeta/gait-recognition/blob/master/human_pose_nn.py#L156). It was used for training purposes to generate desired outputs from coordinates of human's joints. The output is based on type of the loss function, i.e. SCE (sigmoid cross-entropy) or MSE (mean squared error). You can use MSE that just generates 2D gaussians with center at joints location. These maps express the confidence of the potential joint presence for each pixel. The code for generating such maps and identification vector can be as follows

```python
import utils
import numpy as np

# Where count is count of frames and joints is tensor with shape (count, axis, joint_number)
spatial_features = utils.get_gauss_heat_map(
  shape = (count, 289, 289, 16), 
  is_present = np.ones((count, 16)),
  mean = joints, sigma = 1)

identification_vector = net_gait.feed_forward(spatial_features)
```

**Comment by jbkarle** [comment]
Hi Marian, 
Thank you so much !!

**Comment by jbkarle** [comment]
Hello Marian,
Another question: does human-pose-nn regularize coords?. Or it's advisable to do it?

Many thanks again!!

**Comment by jbkarle** [comment]
Sorry I meant normalize data!

**Comment by marian-margeta** [comment]
Sorry, I was wrong, it is a long time ago I trained this network. Network for pose estimation was trained to produce 'confidence maps' as I stated above, but an input to the second network for gait recognition is an output from the [layer](https://github.com/marian-margeta/gait-recognition/blob/master/human_pose_nn.py#L423) that is before transposed convolution (and some gaussian smoothing). As it turned out, the features of pose before up-scaling with transposed convolution was less sparse and overall better. So it is not such straightforward to use your precomputed human pose coords.

**Comment by jbkarle** [comment]
Ok I understand.  That's a problem because then the gait recognition is linked not to the posee vector  (ie COCO format), but is to pose detector previos layers which will be dependent on it's conceptual approach. The reason for me to look at other pose detectors is because I look forward to get the identification vector from a set of video frames (not live processing), but where there are several individual in each frame, and I only want the id_vector just for one. So I was thinking in a different pose detector with tracking capabilities of a target human from frame to frame, ie labeling each individual across frames. Any other ideas compatible with your great work?.

Many Thanks!

**Comment by jbkarle** [comment]
OK, & thanks for the advise!
